kintex_7/genesys2/readme.md

```diff
@@ -0,0 +1,29 @@
+# BlackParrot FPGA Simulation
+This directory contains scripts that will help you setup a quick simulation with BlackParrot on Vivado.
+
+## Tools Requirement:
+Xilinx Vivado (preferably 2019.1 or newer)
+VCS-MX (Optional, 2018 or newer)
+
+## Directory Breakdown
+The bp_packaged_IP contains all of the IP related files while the script directory contains tcl scripts for Vivado. The scripts are broken into 3 parts so that the tool is not doing too much at once.
+
+#### bp_packaged_IP:
+- src: all of the source files needed for BlackParrot IP
+- tb: all of the simulation files for BlackParrot IP
+
+#### script:
+- proj_setup.tcl: sets up a Vivado project and VCS simulator
+- bd_setup.tcl: sets up a BlackParrot block diagram design
```

Comment:
Which of these does the IP export?

Reply:
May you clarify what you meant by IP export?

Reply:
What generates bp_packaged_IP?

Reply:
I've manually generated the IP. All the user have to do with bp_packaged_ip is to say "hey, here is my packaged IP folder, add it to my IP catalog so I can use it in my block design"

It may not make sense for the user to use a script to generate/modify the IP. The user needs to know which files are needed to generate the IPs and how to assign different properties. In the user guide, I go over how to generate and modify the IPs through GUI. To have a script that creates and packages the IP will only be good if the user already know how the IP packaging process work. I'd say it's best for them to go through the user guide process and tutorial docs.

Reply:
Unless you just want a script that generates the current version of the BP, then I can do that. But I'll have to go through the process of creating and packaging the IP to get those tcl commands out.

Reply:
Yeah, there are two use cases being conflated here. 

There are users which want to click download and get the IP. A short guide about how to include it in a custom project is useful. It looks like these are the TCL scripts which are needed for that. 

Then there is us. We want to create the IP automatically every release of master. This should be a TCL script that can be run in a CI job. Uploading the tarbell manually is fine for now, because we can automate that in the future. But we don’t want to require someone to have to manually go through a GUI and select options, because it’s super error-prone.